### PR TITLE
Generic CommitTime Exporter with Webhook - GitHub Support

### PR DESCRIPTION
**Pull request by haithamshahin333**
## Describe the behavior changes introduced in this PR
This PR introduces a webhook endpoint to receive build information that can then be exported to the Pelorus dashboard with the generic exporter. This can be used when builds occur outside of OCP.

## Linked Issues?

related to #246 

## Testing Instructions

deploy to dev cluster using helm charts

Please include any additional commands or pointers in addition to our [standard PR testing process](/docs/Development.md#testing-pull-requests).

@etsauer @shaheinm Further Areas for Improvement:

1. Templates for secrets autogenerate random values - this may be problematic with helm install/upgrade since the secrets would then be overwritten.
2. Would be good to refactor generic exporter such that it is in the committime folder since it leverages same collector classes.
3. Add support for gitlab and bitbucket
4. Requires tests

@redhat-cop/mdt

### CODE REVIEW

**Comment by etsauer** [comment]
@haithamshahin333 to be clear on this. This code only works with CommitTime correct? Its not 'generic' to the metric?

**Comment by haithamshahin333** [comment]
correct @etsauer good callout - I think the pattern could be leveraged for any metric but right now only implemented for CommitTime.

**Comment by haithamshahin333** [comment]
@etsauer added a basic example pipeline in the samples directory that demonstrates usage of this new exporter alongside the deployed webhook. The Jenkinsfile gives an example for how to post build information to the webhook as part of a pipeline.

**Comment by shaheinm** [comment]
Linking this issue here because it's the first instance I see where the promtool tests fail because of this bug - https://github.com/prometheus/prometheus/issues/8494

**Comment by haithamshahin333** [comment]
@etsauer @shaheinm Here are instructions for how to test this new feature with the sample application that is part of the PR (https://github.com/haithamshahin333/pelorus/blob/master/samples/README.md). I added it as a README within the samples directory so that others in the future can describe how to deploy different use cases.

**Review by haithamshahin333 — COMMENTED** [review]
@shaheinm may be an issue with this update, doesn't look like the action was able to start: https://github.com/konveyor/pelorus/actions/runs/573100473

**Comment by shaheinm** [comment]
> @shaheinm may be an issue with this update, doesn't look like the action was able to start: https://github.com/konveyor/pelorus/actions/runs/573100473

Yeah, it looks like there were lots of changes from the version that was being used (a v1 release candidate) and the latest (v2.0.1) for the action.

**Comment by etsauer** [comment]
> > @shaheinm may be an issue with this update, doesn't look like the action was able to start: https://github.com/konveyor/pelorus/actions/runs/573100473
> 
> Yeah, it looks like there were lots of changes from the version that was being used (a v1 release candidate) and the latest (v2.0.1) for the action.

@haithamshahin333 @shaheinm will one of you open an issue on this? We can handle it as a followup.

**Comment by shaheinm** [comment]
> > > @shaheinm may be an issue with this update, doesn't look like the action was able to start: https://github.com/konveyor/pelorus/actions/runs/573100473
> > 
> > 
> > Yeah, it looks like there were lots of changes from the version that was being used (a v1 release candidate) and the latest (v2.0.1) for the action.
> 
> @haithamshahin333 @shaheinm will one of you open an issue on this? We can handle it as a followup.

@etsauer I've already got an update on haitham's master branch in PR that'll make into this one if accepted

File: .github/workflows/prometheus-rules.yml [hunk 1]
```diff
@@ -11,6 +11,6 @@ jobs:
         go-version: '^1.13.1' # The Go version to download (if necessary) and use.
     - run: |
         go version
-        go get github.com/prometheus/prometheus/cmd/promtool
+        GO111MODULE=on go get github.com/prometheus/prometheus/cmd/promtool
```
Comment (roidelapluie) [inline .github/workflows/prometheus-rules.yml:14]:
```suggestion
        GO111MODULE=on go get github.com/prometheus/prometheus/cmd/promtool@release-2.25
```

**Comment by etsauer** [comment]
resolves #246

**Comment by shaheinm** [comment]
I've updated the chart-lint.yml workflow to use the latest version of helm/chart-testing-action - this is necessary as older versions use older version of helm, in which the `lint` command has no knowledge of the `lookup` function in charts (used a bit for this work to ensure k8s secrets are not modified every time a helm update occurs).

This also required that I move the dependency charts (webhooks and exporters) out of the `charts/pelorus/charts` directory and into a `charts/pelorus/subcharts` directory, as helm 3 expects to use the charts directory name for remote dependencies that are downloaded.

**Comment by shaheinm** [comment]
Also updated the conftest action to update dependent charts prior to testing (it runs `helm template` which will fail without the update - should stay as long as the dependency pattern is used)

**Comment by haithamshahin333** [comment]
Replaced by https://github.com/konveyor/pelorus/pull/253

**Pull request closed without merging**